### PR TITLE
Upgrade directory to 3.14.1

### LINKED ISSUE #607: Upgrade Directory to Octane

### PR DESCRIPTION
**Pull request by kevinhinterlong**
Resolves #607 

## Description
- Enable octane for directory

## Proposed Changes
There's a lot of lines changed, but it's easy to read
- Native class syntax
- Angle bracket
- `.ember-cli` -> `.ember-cli.js`
- Breaking: `dir-icon` no longer supports positional params

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/directory/addon/components/dir-asset-row-actions.js [hunk 1]
```diff
@@ -3,50 +3,44 @@
...
 
   /**
    * @property {Object} assetActionComponents - action list component for each asset type
    */
-  assetActionComponents: computed(() => {
-    const components = {
-      report: 'navi-action-list',
-      dashboard: 'dashboard-action-list'
-    };
+  assetActionComponents = {
```
Comment (mattgraham1995) [inline packages/directory/addon/components/dir-asset-row-actions.js]: Is the state leaking issue fixed in native classes?

File: packages/directory/addon/components/dir-item-name-cell.js [hunk 2]
```diff
@@ -16,45 +16,45 @@ import layout from '../templates/components/dir-item-name-cell';
...
 
     return `${pluralType}.${type}`;
-  }),
+  }
 
   /**
    * @property {String} itemId - the id of the model or the tempId of an unsaved model
    */
-  itemId: oneWay('value.modelId'),
+  @oneWay('value.modelId')
```
Comment (mattgraham1995) [inline packages/directory/addon/components/dir-item-name-cell.js]: This should probably be `readOnly`

File: packages/directory/addon/components/dir-asset-row-actions.js (hunk 1 above)
Comment (kevinhinterlong) [reply packages/directory/addon/components/dir-asset-row-actions.js]: Yup, see [Class Fields vs. `extend()`](https://blog.emberjs.com/2019/01/26/emberjs-native-class-update-2019-edition.html)

File: packages/directory/addon/components/dir-icon.js [hunk 3]
```diff
@@ -1,20 +1,19 @@
 /**
- * Copyright 2018, Yahoo Holdings Inc.
+ * Copyright 2019, Yahoo Holdings Inc.
  * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
  *
  * Usage:
- * {{dir-icon iconClass}}
+ * <DirIcon iconClass>
```
Comment (daniellevinson) [inline packages/directory/addon/components/dir-icon.js]: `/>` ?

File: packages/directory/addon/components/dir-search-bar.js [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
  * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
  *
  * Usage:
- * {{dir-search-bar
- *   query=query
- *   searchFor=(action 'searchFor' query)
- * }}
+ * <DirSearchBar
+ *   @query={{query}}
+ *   @searchFor={{action 'searchFor' query}}
```
Comment (daniellevinson) [inline packages/directory/addon/components/dir-search-bar.js]: double quotes 😬

File: packages/directory/addon/components/dir-icon.js (hunk 3 above)
Comment (jkusa) [reply packages/directory/addon/components/dir-icon.js]:
```suggestion
 * <DirIcon "icon-class" />
```

**Pull request merged**